central/auth/service/service_impl.go

```diff
@@ -89,3 +129,137 @@ func authStatusForID(id authn.Identity) (*v1.AuthStatus, error) {
+	resolveGitHubActionsIssuer(config)
+	if err := s.validateAuthMachineToMachineConfig(config, false); err != nil {
+		return nil, err
+	}
+
+	if err := s.authDataStore.UpdateAuthM2MConfig(ctx, v1tostorage.AuthM2MConfig(config)); err != nil {
+		return nil, err
+	}
+
+	return &v1.Empty{}, nil
+}
+
+func (s *serviceImpl) DeleteAuthMachineToMachineConfig(ctx context.Context, id *v1.ResourceByID) (*v1.Empty, error) {
+	if err := s.authDataStore.RemoveAuthM2MConfig(ctx, id.GetId()); err != nil {
+		return nil, errox.InvalidArgs.
+			Newf("could not delete auth machine to machine config with id %q", id.GetId()).CausedBy(err)
+	}
+	return &v1.Empty{}, nil
+}
+
+func (s *serviceImpl) ExchangeAuthMachineToMachineToken(_ context.Context, _ *v1.ExchangeAuthMachineToMachineTokenRequest) (*v1.ExchangeAuthMachineToMachineTokenResponse, error) {
+	return nil, errox.InvariantViolation.New("not yet implemented")
+}
+
+func (s *serviceImpl) validateAuthMachineToMachineConfig(config *v1.AuthMachineToMachineConfig, skipIDCheck bool) error {
+	if config == nil {
+		return errox.InvalidArgs.New("empty config given")
+	}
+	if config.GetId() == "" && !skipIDCheck {
+		return fmt.Errorf("%w: %w", errox.InvalidArgs, errEmptyID)
+	}
+
+	duration, err := time.ParseDuration(config.GetTokenExpirationDuration())
+	if err != nil {
+		return fmt.Errorf("%w: %w: %w", errox.InvalidArgs, errInvalidTokenExpiration, err)
+	}
+
+	if duration < time.Minute || duration > 24*time.Hour {
+		return fmt.Errorf("%w: %w: token expiration must be between 1 minute and 24 hours, but was %s",
+			errox.InvalidArgs, errInvalidTokenExpiration, duration.String())
+	}
+
+	if err := validateIssuer(config); err != nil {
+		return err
+	}
+
+	var regexValidationErrs error
+	for _, mapping := range config.GetMappings() {
+		if mapping.GetValueExpression() == "" {
+			regexValidationErrs = errors.Join(regexValidationErrs,
+				fmt.Errorf("%w for key %q: empty regex given", errInvalidRegularExpression, mapping.GetKey()))
+		}
+		if _, err := regexp.Compile(mapping.GetValueExpression()); err != nil {
+			regexValidationErrs = errors.Join(regexValidationErrs,
+				fmt.Errorf("%w for key %q: %w", errInvalidRegularExpression, mapping.GetKey(), err))
+		}
+	}
+
+	if regexValidationErrs != nil {
+		return fmt.Errorf("%w: %w", errox.InvalidArgs, regexValidationErrs)
+	}
+
+	return nil
+}
+
+func validateIssuer(config *v1.AuthMachineToMachineConfig) error {
+	// For Generic types, the issuer has to be set.
+	if config.GetType() == v1.AuthMachineToMachineConfig_GENERIC && config.GetIssuer() == "" {
+		return fmt.Errorf("%w: %w: type %s was used, but no configuration for the issuer was given",
+			errox.InvalidArgs, errInvalidIssuer, config.GetType())
+	}
+	// For GitHub action types, the issuer either has to be empty or set to the github actions issuer.
+	if config.GetType() == v1.AuthMachineToMachineConfig_GITHUB_ACTIONS &&
+		(config.GetIssuer() != githubActionsIssuer && config.GetIssuer() != "") {
```

Comment:
Since we know that `resolveGitHubActionsIssuer` has to be called beforehand and since we should pass empty issuer further or otherwise exchanging GHA tokens will fail.
```suggestion
	// For GitHub action types, there's only one valid issuer value.
	if config.GetType() == v1.AuthMachineToMachineConfig_GITHUB_ACTIONS &&
		config.GetIssuer() != githubActionsIssuer {
```